pkgs/applications/editors/vim/plugins/overrides.nix

```diff
@@ -1281,6 +1281,7 @@ in
     dependencies = [ self.plenary-nvim ];
   };
 
+  # NOTE: this overrides a luaPackages-based plugin
   gitsigns-nvim = super.gitsigns-nvim.overrideAttrs {
```

Comment:
the main driver for using `buildNeovimPlugin` is to not have to add dependencies. Could this have been deleted instead ? same remark for the rest.

Reply:
Nixvim's update has been blocked for a while because these overrides were no longer being applied (they we're being silently shadowed by the luaPackages-based plugins). So at least currently, most of them are still needed.

I agree, ideally `buildNeovimPlugin` and/or `toVimPackage` would correctly propagate the underlying lua package's dependencies, however it seems like that isn't currently happening.

It's also possible that the dependencies _are_ being propagated, but there is an issue in nixvim's combine-plugins impl. IIRC we only saw errors when combine-pluigins was being used.

Either way, overrides shouldn't be silently ignored, and this PR fixes _that_.